src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -117,6 +123,24 @@ internal class CertificateManager
             bool HasOid(X509Certificate2 certificate, string oid) =>
                 certificate.Extensions.OfType<X509Extension>()
                     .Any(e => string.Equals(oid, e.Oid.Value, StringComparison.Ordinal));
+
+            bool MatchesVersion(X509Certificate2 c)
+            {
+                var byteArray = c.Extensions.OfType<X509Extension>()
+                                    .Where(e => string.Equals(AspNetHttpsOid, e.Oid.Value, StringComparison.Ordinal))
+                                    .Single()
+                                    .RawData;
+                if (byteArray.Length == AspNetHttpsOidFriendlyNameLength)
```

Comment:
nit: new line before the if
nit: don't tab new lines on chained expressions all the way to the right. A single tab is enough. Keep the style consistent with that is already here.

Reply:
This is the reason we need analyzers/code checks for style 😄